src/main/java/uk/gov/hmcts/ccd/domain/service/search/CaseSearchResultViewGenerator.java

```diff
@@ -220,7 +219,13 @@ private SearchResultViewItem buildSearchResultViewItem(CaseDetails caseDetails,
         });
 
         newResults.putAll(caseData);
-        newResults.putAll(metadata);
+        newResults.putAll(convertReferenceToString(metadata));
         return newResults;
     }
+
+    private Map<String, Object> convertReferenceToString(Map<String, Object> caseMetadata) {
+        final String convertedCaseReference = String.valueOf(caseMetadata.get(CASE_REFERENCE.getReference()));
+        caseMetadata.put(CASE_REFERENCE.getReference(), convertedCaseReference);
```

Comment:
a minor, not sure if it could cause any issue without a deeper investigation:
you are modifying the original map in CaseDetails here. Perhaps safer to make a copy of the map `caseMetadata` and return the copy with the modified case reference.
Or, not modify the Map, but directly modifying `newResults` in the caller.

Reply:
created copy of the map